modules/citrus-integration/src/it/java/com/consol/citrus/javadsl/runner/GlobalVariablesTestRunnerIT.java

```diff
@@ -0,0 +1,95 @@
+import org.testng.annotations.Optional;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+/**
+ * Tests the loading and usage of global variables in citrus
+ */
+@Test
+public class GlobalVariablesTestRunnerIT extends TestNGCitrusTestRunner {
+
+    @CitrusTest
+    public void globalProperties() {
+        echo("Project name is: ${project.name}");
+
+        echo("Testing global variables from properties: ${globalWelcomingText}");
+    }
+
+    @CitrusTest
+    @Parameters({"runner", "context"})
+    public void randomLocalVariable_readMultipleTimes_valueShouldRemainConstantOnEachRead(
+            @Optional @CitrusResource TestRunner runner, @Optional @CitrusResource TestContext context) {
+
+        runner.variable("localRandomNumber1", "citrus:randomNumber(5)");
+        runner.echo("localRandomNumber1: ${localRandomNumber1}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "localRandomNumber1");
+
+        runner.createVariable("localRandomNumber2", "citrus:randomNumber(8)");
+        runner.echo("localRandomNumber2: ${localRandomNumber2}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "localRandomNumber2");
+    }
+
+    @CitrusTest
+    @Parameters({"runner", "context"})
+    public void randomGlobalVariable_readMultipleTimes_valueShouldRemainConstantOnEachRead(
+            @Optional @CitrusResource TestRunner runner, @Optional @CitrusResource TestContext context) {
+        runner.echo("globalRandomNumber1: ${globalRandomNumber1}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "globalRandomNumber1");
+    }
+
+    @CitrusTest
+    @Parameters({"runner", "context"})
+    public void multipleGlobalVariablesWithCrossReferencesAndFunctions_readMultipleTimes_valueShouldRemainConstantOnEachRead(
+            @Optional @CitrusResource TestRunner runner, @Optional @CitrusResource TestContext context) {
+        runner.echo("globalSum1: ${globalSum1}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "globalSum1");
+
+        runner.echo("globalSum2: ${globalSum2}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "globalSum2");
+
+        assertVariableValuesMatch(context, "globalSum1", "globalSum2");
+    }
+
+    private void assertVariableValuesMatchWhenReadMultipleTimes(final TestContext testContext, final String name) {
+        String val1 = testContext.getVariable(name);
+        String val2 = testContext.getVariable(name);
+        if (val1.equals(val2)) {
+            log.debug(String.format("Values match for variable %s. Value: %s", name, val1));
+        } else {
+            throw new RuntimeException(String.format("Values don't match for variable %s. Value1: %s, Value2: %s", name, val1, val2));
+        }
+    }
+
+    private void assertVariableValuesMatch(final TestContext testContext, final String name1, final String name2) {
+        String val1 = testContext.getVariable(name1);
```

Comment:
Same issue as for `assertVariableValuesMatchWhenReadMultipleTimes`

```suggestion
        String val1 = testContext.resolveDynamicValue(testContext.getVariable(name1));
```